.github/dependabot.yml

```diff
@@ -31,3 +39,8 @@ updates:
       all:
         patterns:
         - "*"
+    allow:
+      - dependency-name: "pypa/hatch"
+        versions: [ "install" ] # stay on "install" branch
+      - dependency-name: "github/codeql-action"
+        versions: [ ">=4.0.0" ] # stay on "v4" tag (and "v5" in the future)
```

Comment:
not personally familiar with, or see the necessity of this. But at the same time, dependabot isn't currently working for github/codeql-action here and I don't know why. So not opposed to it